### DISCUSSION TITLE
Using OWL 2 OML adapter

### DISCUSSION THREAD

**Discussion by raks0009** [opencaesar/platform#28] [Q&A] [answered]
Hey there, I'm trying to use the OWL 2 OML adapter to be able to use some ontologies provided on the web.
Specifically, I am interested in using the QUDT ontology (https://qudt.org)
My Rosetta-based OML project is available at: https://gitlab.rakshitmittal.net/mde/msdl/adamanthea

I tried adding the corresponding task to build.gradle. It does run, but I see no output.
What would possible options be to able to import such foreign ontologies that are not provided in OML core/metrology vocabularies?

It would be great if a demo for something like this could be provided., just like other OML capabilities.

Thanks!

**Comment by melaasar** [comment, marked as answer]
Hi and sorry for the delay of getting back to you.

Currently, the owl2oml adapter supports two levels of importing OWL based on the used profile:
a) When the ontologies use the OML profile (like what you get then the oml2owl is run)
b) When the ontologies do not use the OML profile

Currently, what is productized is (a). It should work as described in the README. There is no tutorial yet but there is a [branch of the Kepler16b example](https://github.com/opencaesar/kepler16b-example/blob/8af0eb52258044818d1b3cb09c83f652566c752f/build.gradle#L190-L196) that shows this.

The (b) level is still work in progress though. I did not release it yet, but I could do it on a provisional (experimental) basis if you like. Otherwise to use (b), you could get the source, modify [one line](https://github.com/opencaesar/owl-adapter/blob/6375ba232a2a9a9ee8a0d395e05112bda2b1f205/owl2oml/src/main/java/io/opencaesar/owl2oml/Owl2OmlApp.java#L251) (replace Owl2Oml by Owl2OmlEx). Then either publish it to local Maven (./gradlew publishToMavenLocal) and use it from Gradle, or run it as a java app directly.
